core/embed/util/image/boot_header.c

```diff
@@ -0,0 +1,322 @@
+/*
+ * This file is part of the Trezor project, https://trezor.io/
+ *
+ * Copyright (c) SatoshiLabs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifdef SECURE_MODE
+
+#include <trezor_model.h>
+#include <trezor_rtl.h>
+
+#include <rtl/sizedefs.h>
+#include <util/boot_header.h>
+#include <util/image_hash_conf.h>
+
+#include <../vendor/sphincsplus/ref/api.h>
+#include <ed25519-donna/ed25519.h>
+
+#include <version.h>
+
+#ifdef BOOTLOADER
+extern const uint8_t _bootloader_code_size;
+
+typedef union {
+  boot_header_auth_t hdr;
+  uint8_t raw[BOOT_HEADER_MAXSIZE];
+} boot_header_padded_t;
+
+__attribute__((section(".header")))
+const boot_header_padded_t g_bootloader_header = {
+    .hdr = {
+        .magic = BOOT_HEADER_MAGIC_TRZQ,
+        .hw_model = HW_MODEL,
+        .hw_revision = HW_REVISION,
+        .version =
+            {
+                .major = VERSION_MAJOR,
+                .minor = VERSION_MINOR,
+                .patch = VERSION_PATCH,
+                .build = VERSION_BUILD,
+            },
+        .fix_version =
+            {
+                .major = FIX_VERSION_MAJOR,
+                .minor = FIX_VERSION_MINOR,
+                .patch = FIX_VERSION_PATCH,
+                .build = FIX_VERSION_BUILD,
+            },
+        .min_prev_version =
+            {
+                .major = 0,
+                .minor = 0,
+                .patch = 0,
+                .build = 0,
+            },
+        .monotonic_version = BOOTLOADER_MONOTONIC_VERSION,
+        .sigmask = 0,  // This field will be set by headertool_pq
+        .header_size = BOOT_HEADER_MAXSIZE,
+        .auth_size = BOOT_HEADER_MAXSIZE - sizeof(boot_header_merkle_proof_t) -
+                     sizeof(boot_header_unauth_t),
```

Comment:
How does this work? `sizeof(boot_header_merkle_proof_t)` does not include the size of the nodes, it's just the `uint32_t node_count`. In fact we don't necessarily know the node count at compile time. So shouldn't this field be set like `sigmask` by headertool?